src/Symfony/Bundle/WebProfilerBundle/Tests/DependencyInjection/WebProfilerExtensionTest.php

```diff
@@ -125,70 +117,13 @@ public function testToolbarConfig(bool $toolbarEnabled, bool $listenerInjected,
         }
     }
 
-    public function getToolbarConfig()
-    {
-        return [
-            [
-                'toolbarEnabled' => false,
-                'listenerInjected' => false,
-                'listenerEnabled' => false,
-            ],
-            [
-                'toolbarEnabled' => true,
-                'listenerInjected' => true,
-                'listenerEnabled' => true,
-            ],
-        ];
-    }
-
-    /**
-     * @group legacy
-     *
-     * @dataProvider getInterceptRedirectsToolbarConfig
-     */
-    public function testToolbarConfigUsingInterceptRedirects(
-        bool $toolbarEnabled,
-        bool $interceptRedirects,
-        bool $listenerInjected,
-        bool $listenerEnabled
-    ) {
-        $extension = new WebProfilerExtension();
-        $extension->load(
-            [['toolbar' => $toolbarEnabled, 'intercept_redirects' => $interceptRedirects]],
-            $this->container
-        );
-        $this->container->removeDefinition('web_profiler.controller.exception');
-
-        $this->assertSame($listenerInjected, $this->container->has('web_profiler.debug_toolbar'));
-
-        self::assertSaneContainer($this->getCompiledContainer(), '', ['web_profiler.csp.handler']);
-
-        if ($listenerInjected) {
-            $this->assertSame($listenerEnabled, $this->container->get('web_profiler.debug_toolbar')->isEnabled());
-        }
-    }
-
-    public function getInterceptRedirectsToolbarConfig()
+    public function getDebugModes()
     {
         return [
-             [
-                 'toolbarEnabled' => false,
-                 'interceptRedirects' => true,
-                 'listenerInjected' => true,
-                 'listenerEnabled' => false,
-            ],
-            [
-                'toolbarEnabled' => false,
-                'interceptRedirects' => false,
-                'listenerInjected' => false,
-                'listenerEnabled' => false,
-            ],
-            [
-                'toolbarEnabled' => true,
-                'interceptRedirects' => true,
-                'listenerInjected' => true,
-                'listenerEnabled' => true,
-            ],
+            [false, false, false, false],
+            [true,  false, true,  true],
+            [false, true,  true,  false],
+            [true,  true,  true,  true],
```

Comment:
Can we at least keep the "good parts" like this?

Reply:
Sure! I'll do that ASAP

Reply:
friendly ping @lyrixx

Reply:
Wep, Sorry I was AFK.

I re-reverted everything in the test folder